### PR TITLE
Add a "batch" and "verbose" mode

### PR DESCRIPTION
**Pull request by Feandil**
The batch mode is to allow it to run on large clusters and simply grep the output.
The verbose mode is to remove some information from the default run, as it seems a bit too much

This patchset currently cannot be merged, as it was written before the nocolor patch.
I'm trying to merge it, but don't have time right now and not sure what would be the best approach.

### CODE REVIEW

**Comment by Feandil** [comment]
2 different conflicts fixed (no-color & ARM)

The is_cpu_vulnerable could be put into pvulnstatus, but I'm tired of rebasing ;)

File: spectre-meltdown-checker.sh [hunk 1]
```diff
@@ -119,31 +33,68 @@ show_usage()
 
 	Options:
 		--no-color			Don't use color codes
-
```
Comment (speed47) [inline spectre-meltdown-checker.sh]: Tiny detail: this \n was actually here for a reason: `./spectre-meltdown-checker.sh --invalid` is more readable in that case

**Comment by speed47** [comment]
./spectre-meltdown-checker.sh: 252: ./spectre-meltdown-checker.sh: Bad substitution

=> Avoid including bashisms, I would like to keep the shell POSIX-compliant (so that it works even where bash is not available)

File: spectre-meltdown-checker.sh [hunk 2]
```diff
@@ -250,16 +200,128 @@ while [ -n "$1" ]; do
...
+	[ -n "$3" ] && _info_nol " ($3)"
+	_info
+}
+
+# Print the final status of a vulnerability (incl. batch mode)
+# Arguments are: CVE UNK/OK/VULN description
+pvulnstatus()
+{
+        [ "$opt_batch" = 1 ] && _echo 0 "$1: $2 ($3)"
+        _info "> \033[46m\033[30mSTATUS:\033[0m "
```
Comment (speed47) [inline spectre-meltdown-checker.sh]: _info_nol

File: spectre-meltdown-checker.sh [hunk 3]
```diff
@@ -273,12 +335,12 @@ fi
...
-		_echo "\033[31mWe'll proceed but you might see permission denied errors.\033[0m"
-		_echo "\033[31mTo run it as root, you can try the following command: sudo $0\033[0m"
-		_echo
+		_warn "Note that you should launch this script with root privileges to get accurate information."
+		_warn "We'll proceed but you might see permission denied errors."
+		_warn "To run it as root, you can try the following command: sudo $0"
+		_warn
 	fi
-	_echo "Checking for vulnerabilities against live running kernel \033[35m"$(uname -s) $(uname -r) $(uname -v) $(uname -m)"\033[0m"
+	_verbose "Checking for vulnerabilities against live running kernel \033[35m"$(uname -s) $(uname -r) $(uname -v) $(uname -m)"\033[0m"
```
Comment (speed47) [inline spectre-meltdown-checker.sh]: I agree with the fact that the messages `Will use vmlinux/sysmap/kconfig ...` can require a `-v`

However I think we should keep the `uname -a` in the default output: when people copy/paste the output from the script, it'll then be of some use to others. Without the `uname -a`, the context is lost and the script output becomes useless except if you're the one who launched it.

For a shorter output, we could split `Checking for vulnerabilities against live running kernel` (and show it only under -v). And keep the actual `uname -a` by default

File: spectre-meltdown-checker.sh [hunk 4]
```diff
@@ -305,24 +367,24 @@ if [ "$opt_live" = 1 ]; then
 		opt_config=/boot/config-$(uname -r)
 	fi
 else
-	_echo "Checking for vulnerabilities against specified kernel"
+	_verbose "Checking for vulnerabilities against specified kernel"
```
Comment (speed47) [inline spectre-meltdown-checker.sh]: As per above, please keep this in standard output too, for context in copy/pastes

**Comment by Feandil** [comment]
New version with bugfixes squashed. Not squashed commits available here: https://github.com/Feandil/spectre-meltdown-checker/compare/1582b65...batch_nosquash

**Comment by speed47** [comment]
Looks good to me, geting some guys to test it here, just to be sure

**Pull request merged**